presto-spi/src/main/java/com/facebook/presto/spi/relation/PredicateCompiler.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.relation;
+
+import java.util.function.Supplier;
+
+public interface PredicateCompiler
+{
+    /**
+     * Predicate expression may not contain any variable references, only input references.
+     */
+    Supplier<Predicate> compilePredicate(RowExpression predicate);
```

Comment:
As discussed offline, let's put a javadoc to denote a few things:
- this is an experimental API
- the goal is do to filter reordering in connector to workaround lazy block loading everything
- we should explore the possibility having a new interface in lazy block (or maybe new block implementation) to selectively load blocks

cc: @arhimondr @wenleix @tdcmeehan

Reply:
@highker Added javadoc to clarify that this is an experimental API.

```
/**
 * @apiNote An experimental API to allow Hive connector to implement smart
 * filtering on the encoded data to avoid materializing data unnecessarily.
 */
public interface PredicateCompiler
{
```